File: app/src/main/java/ru/example/newsreader/MainActivity/MainActivity.java
package ru.example.newsreader.MainActivity;

import android.arch.persistence.room.Room;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.support.v7.widget.Toolbar;
import android.view.Menu;
import android.view.MenuItem;

import com.facebook.stetho.Stetho;

import java.util.List;

import ru.example.newsreader.R;
import ru.example.newsreader.models.Article;
import ru.example.newsreader.room.AppDatabase;
import ru.example.newsreader.room.Migrations;

public class MainActivity extends AppCompatActivity implements MainActivityView{

    private MainActivityPresenter presenter;

    private RecyclerView mRecyclerView;
    private RecyclerView.Adapter mAdapter;
    private RecyclerView.LayoutManager mLayoutManager;
    private AppDatabase database;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        Stetho.initializeWithDefaults(this);

        database = Room.databaseBuilder(getApplicationContext(),
                AppDatabase.class, "rss2-database")
                .addMigrations(Migrations.MIGRATION_1_2, Migrations.MIGRATION_2_3, Migrations.MIGRATION_3_4,
                        Migrations.MIGRATION_4_5)
                .fallbackToDestructiveMigration()
                .build();

        presenter = new MainActivityPresenter(this, database);

        mRecyclerView = (RecyclerView) findViewById(R.id.recycler_view);

        mLayoutManager = new LinearLayoutManager(this);
        mRecyclerView.setLayoutManager(mLayoutManager);

        presenter.getArticlesInSource();
        presenter.downloadArticles();
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu_activity_main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        int id = item.getItemId();
        if (id == R.id.remove_db) {
            presenter.deleteArticlesFromDB();
            return super.onOptionsItemSelected(item);
        }
        return super.onOptionsItemSelected(item);
    }

    @Override
    public void showArticles(List<Article> articleList) {
        if(articleList!=null){
            mAdapter = new RVAdapter(articleList);
            mRecyclerView.setAdapter(mAdapter);
        }
    }
}

File: app/src/main/java/ru/example/newsreader/screens/main_activity/MainActivity.kt
package ru.example.newsreader.screens.main_activity

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.support.v4.widget.ContentLoadingProgressBar
import android.support.v7.widget.AppCompatTextView
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.support.v7.widget.Toolbar
import android.view.Menu
import android.view.MenuItem
import android.view.View
import android.widget.LinearLayout
import com.facebook.stetho.Stetho
import dagger.android.support.DaggerAppCompatActivity
import ru.example.newsreader.R
import ru.example.newsreader.models.Article
import ru.example.newsreader.screens.main_activity.adapter.ArticlesAdapter
import javax.inject.Inject

fun startMainActivity(context: Context) {
    val intent = Intent(context, MainActivity::class.java)
    context.startActivity(intent)
}

interface MainActivityView{
    fun showArticles(articleList: List<Article>)

    fun showProgressBar()

    fun hideProgressBar()

    fun showNoArticlesText()

    fun initViews()
}

class MainActivity :  DaggerAppCompatActivity(), MainActivityView {

    @Inject
    lateinit var presenter : MainActivityPresenter

    private var mRecyclerView: RecyclerView? = null
    private var mAdapter: ArticlesAdapter? = null
    private var mLayoutManager: RecyclerView.LayoutManager? = null

    private var mProgressBar : ContentLoadingProgressBar? = null
    private var mArticlesView: LinearLayout? = null
    private var mNoArticlesText: AppCompatTextView? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val toolbar : Toolbar = findViewById(R.id.toolbar)
        setSupportActionBar(toolbar)

        Stetho.initializeWithDefaults(this)

        initViews()
    }

    override fun hideProgressBar() {
        mProgressBar?.visibility = View.GONE
        mArticlesView?.visibility = View.VISIBLE
    }

    override fun showProgressBar() {
        mArticlesView?.visibility = View.GONE
        mProgressBar?.visibility = View.VISIBLE
    }

    override fun showNoArticlesText() {
        mNoArticlesText?.visibility = View.VISIBLE
    }


    override fun initViews() {
        mArticlesView=findViewById(R.id.articles_view)
        mProgressBar = findViewById(R.id.progressbar)
        mNoArticlesText = findViewById(R.id.no_articles_text)

        mRecyclerView = findViewById(R.id.recycler_view)
        mLayoutManager = LinearLayoutManager(this)
        mRecyclerView?.layoutManager = mLayoutManager
    }

    override fun showArticles(articleList: List<Article>) {
        mAdapter = ArticlesAdapter(articleList, this)
        mRecyclerView?.adapter = mAdapter
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when(item.itemId){
            R.id.remove_db -> {
                presenter.deleteArticlesFromDB()
            }
        }
        return super.onOptionsItemSelected(item)
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        menuInflater.inflate(R.menu.menu_activity_main, menu)
        return super.onCreateOptionsMenu(menu)
    }
}